src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -137,6 +137,14 @@ class PgAutoscaler(MgrModule):
                        'means starts out with full pgs and scales down when '
                        'there is pressure '),
             runtime=True),
+        Option(
+            name='threshold',
+            type='float',
+            desc='scaling threshold',
+            long_desc=('The factor by which the `NEW PG_NUM` must vary from the current'
+                       '`PG_NUM` before being accepted. Should not be less than 2.0'),
```

Comment:
In the spirit of both accuracy and helpfulness, I would suggest s/'Should not be less than 2.0'/'Cannot be less than 2.0 given that the autoscaler will always choose powers of 2 for `NEW PG_NUM`'

Having said that, what is the behaviour when the current `PG_NUM` is not a power of 2, e.g. if the user did something funky? I wonder if it's actually worth allowing `min=1.1` or something like that...

Reply:
I made a [spreadsheet](https://docs.google.com/spreadsheets/d/1TJRx-R98qaNILAzgSQiCiCnr6tH2-nSIVxiOftp12ek/edit?usp=sharing) to visualize scaling up with an adjustable threshold. Seems like things shouldn't really get weird until the threshold is less than 1.0 at which point it might enter a infinite loop because it will be acceptable to "scale" without actually changing the number of PGs. Maybe I'm not accounting for some other behavior thought that determined the original 2.0 assertion.

Reply:
Looks like the threshold was [moved from 1.0 to 2.0 in the past](https://github.com/ceph/ceph/commit/0500020613cc7b0b36b3104db05da52321806c1f)

@kamoltat Could you share the reasoning?

Reply:
@sosodev 
Hi so actually the commit message in https://github.com/ceph/ceph/commit/0500020613cc7b0b36b3104db05da52321806c1f
was not clear. I accidentally changed the threshold to 1.0 for testing purposes and so I changed it back to the original value which was 3.0 and assert  >= 2.0. 

To be honest, I am unsure of the exact reason why it is 2.0 for the assertion, but if had to guess it is because we don't want the pools to scale too frequently.

Maybe @liewegas or @jdurgin can give a better explanation.

Reply:
@kamoltat I think I figured it out. Before your scaling profile change the autoscaler could scale up or down as long as the NEW PG_NUM exceeded the threshold. So at that point setting the threshold below 2.0 was a bad idea because it meant that any NEW PG_NUM would be acceptable and scaled to without hesitation. 

So in the versions of Ceph before the profile change the threshold acts as a buffer that prevents scaling in either direction until the difference is quite large. In versions with the profile change the threshold simply reduces how often the autoscaler will scale in the predefined direction and can thus be reduced quite a bit.